### PR TITLE
Support MemmappedFileSystem

### LINKED ISSUE #1231: Moving to MemmappedEnv file loading
Yes we RAM.

### PR DESCRIPTION
**Pull request by lissyx**
Fixes #1231

### CODE REVIEW

File: native_client/deepspeech.cc [hunk 1]
```diff
@@ -56,13 +59,36 @@ Model::Model(const char* aModelPath, int aNCep, int aNContext,
     return;
   }
 
-  Status status = NewSession(SessionOptions(), &mPriv->session);
+  Status status;
+  SessionOptions options;
+  bool is_mmap = std::string(aModelPath).find(".pbmm") != std::string::npos;
```
Comment (lissyx) [inline native_client/deepspeech.cc]: Simple way to easily distinguish between both. Trying to use the wrong file will fail, anyway.

**Comment by lissyx** [comment]
@reuben This should work good, and handle both files. I have not yet updated the tests, because I'm wondering whether we should convert all tests to use `.pbmm` files only, or test both ?

**Comment by lissyx** [comment]
This will have to wait on https://github.com/mozilla/tensorflow/pull/53

File: native_client/deepspeech.cc [hunk 2]
```diff
@@ -56,13 +59,39 @@ Model::Model(const char* aModelPath, int aNCep, int aNContext,
...
+  if (!is_mmap) {
+    std::cerr << "Warning: this model not mmap(). It will consume much more heap memory." << std::endl;
```
Comment (reuben) [inline native_client/deepspeech.cc]: Maybe rephrase this message as "Warning: reading entire model file into memory. Transform model file into an mmapped graph to reduce heap usage."

File: native_client/deepspeech.cc [hunk 3]
```diff
@@ -42,6 +44,7 @@ Model::Model(const char* aModelPath, int aNCep, int aNContext,
              const char* aAlphabetConfigPath, int aBeamWidth)
 {
   mPriv             = new Private;
+  mPriv->mmap_env   = new MemmappedEnv(Env::Default());
```
Comment (reuben) [inline native_client/deepspeech.cc]: This is only used in this constructor, so there's no need to put it in `Private`.

Comment (lissyx) [reply native_client/deepspeech.cc]: We actually need to hold the pointer for the duration of `Model` :)
